pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1234,12 +1235,24 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 
 // getUnschedulablePodsWithMatchingAffinityTerm returns unschedulable pods which have
 // any affinity term that matches "pod".
+//
+// This function is supposed to get called when AssignedPodAdded or AssignedPodUpdated happen
+// so that we can reduce the number of pods evaluated in a further process.
+// We know it depends on the logic of InterPodAffinity plugin, which is not ideal since the scheduling queue should be pure,
+// and after all QueueingHint work finishes, we'll cleanup this function along with PreEnqueueCheck.
+//
 // NOTE: this function assumes lock has been acquired in caller.
 func (p *PriorityQueue) getUnschedulablePodsWithMatchingAffinityTerm(logger klog.Logger, pod *v1.Pod) []*framework.QueuedPodInfo {
 	nsLabels := interpodaffinity.GetNamespaceLabelsSnapshot(logger, pod.Namespace, p.nsLister)
 
 	var podsToMove []*framework.QueuedPodInfo
 	for _, pInfo := range p.unschedulablePods.podInfoMap {
+		if pInfo.Gated && pInfo.UnschedulablePlugins.Has(names.SchedulingGates) {
```

Comment:
It takes the same approach as https://github.com/kubernetes/kubernetes/pull/125527, using `pInfo.UnschedulablePlugins.Has` to make this shortcut available only for the scheduling gate plugin.